test/integration/test_integration__io.py

```diff
@@ -1236,3 +1236,52 @@ def test_cint16_promote_dtype(tmp_path):
         assert "complex128" in riofh.dtypes
         assert riofh.nodata == 0
         assert data.dtype == "complex128"
+
+
+def test_reading_gcps(tmp_path):
+    """
+    Test reading gcps from a tiff file.
+    """
+    tiffname = tmp_path / "test.tif"
+    src_gcps = [
+        GroundControlPoint(
+            row=0, col=0, x=0.0, y=0.0, z=12.0, id="1", info="the first gcp"
+        ),
+        GroundControlPoint(
+            row=0, col=800, x=10.0, y=0.0, z=1.0, id="2", info="the second gcp"
+        ),
+        GroundControlPoint(
+            row=800, col=800, x=10.0, y=10.0, z=3.5, id="3", info="the third gcp"
+        ),
+        GroundControlPoint(
+            row=800, col=0, x=0.0, y=10.0, z=5.5, id="4", info="the fourth gcp"
+        ),
+    ]
+    crs = CRS.from_epsg(4326)
+    with rasterio.open(
+        tiffname,
+        mode="w",
+        height=800,
+        width=800,
+        count=3,
+        dtype=np.uint8,
+        driver="GTiff",
+    ) as source:
+        source.gcps = (src_gcps, crs)
+
+    with rioxarray.open_rasterio(tiffname) as darr:
+        assert "gcps" in darr.attrs
+        assert "gcp_crs" in darr.attrs
+        assert darr.attrs["gcp_crs"] == crs
+        gcps = darr.attrs["gcps"]
+        assert gcps["type"] == "FeatureCollection"
+        assert len(gcps["features"]) == len(src_gcps)
```

Comment:
I would be good to verify that it will be safely written to a netCDF using `to_netcdf`. Anything loaded in using `open_rasterio` shouldn't cause serialization issues.

Reply:
Good point, I hadn't though of that. This is still very preliminary though as you can see, just reading is implemented. To be continued :)

Reply:
BTW, how do we implement GCPs in netcdf ?

Reply:
This may mean that the attribute may have to be a string.

Reply:
From here: https://github.com/corteva/rioxarray/issues/376#issue-938976551

I think `rio.get_gcps()` and a `rio.write_gcps()` would be useful methods to add. This way, the user doesn't have to be concerned about the geojson and only interfaces with ` GroundControlPoint` objects just like they would if they were using rasterio.

Reply:
Sure, let me see if I can find where to implement this.

Reply:
Here is the `rio.transform`|`rio.write_transform` methods for reference: [link](https://github.com/corteva/rioxarray/blob/1f3ed6c5db2475a0d9b9aa9c4985c1b0c558c6e6/rioxarray/rioxarray.py#L522-L596)

Reply:
perfect, thanks